### PR TITLE
Add negative scopes for all enum values

### PR DESCRIPTION
**Pull request by dhh**
```ruby
class Post < ActiveRecord::Base
  enum status: %i[ drafted active trashed ]
end

Post.not_drafted # => where.not(status: :drafted)
Post.not_active  # => where.not(status: :active)
Post.not_trashed # => where.not(status: :trashed)
```

Particularly convenient when you want to disallow access in a controller, like `Post.not_trashed.find(params[:id])`.

### CODE REVIEW

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
+*   Add negative scopes for all enum values.
+
+    Example:
+    
+      class Post < ActiveRecord::Base
```
Comment (abhaynikam) [inline activerecord/CHANGELOG.md]: @dhh: I think you need to add extra 2 spaces here. The example structure is broken here https://github.com/rails/rails/blob/negative-enum-scope/activerecord/CHANGELOG.md

Comment (abhaynikam) [reply activerecord/CHANGELOG.md]: Also, I don't know if it matters or not, but we should inherit class `Post` from `ApplicationRecord` instead here

**Comment by abhaynikam** [comment]
@dhh : I think we should also add `post.not_active?` like we already have `post.active?`. What do you think?

File: activerecord/CHANGELOG.md (hunk 1 above)
Comment (dhh) [reply activerecord/CHANGELOG.md]: I'll do the indent, but ApplicationRecord is really a Rails concern, not an Active Record concern. Maybe a fine point of difference, but it's there.

**Comment by dhh** [comment]
Not needed. You can always negate a predicate method in a variety of ways with `!`, `unless`, and a million other ways. Not so with a scope, thus the need for the help there.

**Comment by brenogazzola** [comment]
This feature will cause some annoyance in projects with a model that declares an enum where one of its members is the negative of the other, such as:

```
class Rsvp < ApplicationRecord
  enum status: [:going, :maybe, :not_going]
end
```

The `enum` generates a scope `:not_going` that returns a different result from the new negative scope for `:going`. Right now the enum scope is the one overwriting so it won't break existing code, but Rails keeps complaining that one scope is overwriting the other:

`WARN -- : Creating scope :not_going. Overwriting existing method Rsvp.not_going.`

Not sure how this should be handled, but suppressing the warning doesn't seem like a good idea, since future changes might cause the negative scope to overwrite the enum scope. And if a developer decides to rename `:not_going` he might introduce bugs because pieces of code that relied on that scope will continue working (and maybe not failing certain tests), but returning a different result.

**Pull request merged**